src/routes/index.js

```diff
@@ -12,23 +8,21 @@ const MAX_LEN = 7;
 
 const v = new Vault();
 
-module.exports = micro(async (req, res) => {
+module.exports = async (req /*: $FlowFixMe */, res /*: $FlowFixMe */) => {
   try {
     let LENGTH /*: number */ = 4;
     let SEPARATOR = "-";
 
-    const { query } = parse(req.url);
-
-    if (query) {
-      if (query.w) {
-        LENGTH = Number(query.w);
+    if (req.query) {
```

Comment:
Sorry to be late here, but just a small tip : `req.query` is `{}` by default so you don't need to check that `req.query` is not undefined 🙂

(cool to see that you're using helpers by the way 🙏)